overviews/core/_posts/2012-09-20-futures.md

```diff
@@ -36,6 +36,134 @@ in the case of an application which requires blocking futures, for an underlying
+
+The parallelism level will be set to `numThreads` as long as it remains within `[minThreads; maxThreads]`.
+
+As stated above the `ForkJoinPool` can increase the amount of threads beyond its `parallelismLevel` in the presence of blocking computation. As explained in the `ForkJoinPool` API, this is only possible if the pool is explicitly notified:
+
+    import scala.concurrent.Future
+    import scala.concurrent.forkjoin._
+
+    // the following is equivalent to `implicit val ec = ExecutionContext.global`
+    import ExecutionContext.Implicits.global
+
+    Future {
+      ForkJoinPool.managedBlock(
+        new ManagedBlocker {
+           var done = false
+
+           def block(): Boolean = {
+             try {
+               myLock.lock()
+               // ...
+             } finally {
+              done = true
+             }
+             true
+           }
+
+           def isReleasable: Boolean = done
+        }
+      )
+    }
+
+
+Fortunately the concurrent package provides a convenient way for doing so:
+
+    import scala.concurrent.Future
+	  import scala.concurrent.blocking
+
+    Future {
+      blocking {
+        myLock.lock()
+        // ...
+      }
+    }
+
+Note that `blocking` is a general construct that will be discussed more in depth [below](#in_a_future).
+
+Last but not least, you must remember that the `ForkJoinPool` isn't designed for long lasting blocking operations. Even when notified with `blocking` the pool might not spawn new workers as you would expect, and when new workers are created they can be as many as 32767. 
+To give you an idea, the following code will use 32000 threads:
+
+    implicit val ec = ExecutionContext.global
+
+    for( i <- 1 to 32000 ) {
+      Future {
+        blocking {
+          Thread.sleep(999999)		
+        }
+      }
+    }
+
+
+If you need to wrap long lasting blocking operations we recommend using a dedicated `ExecutionContext`, for instance by wrapping a Java `Executor`.
+   
+
+### Adapting Java Executor
+
+Using `ExecutionContext.fromExecutor` you can adapt a Java `Executor` into an `ExecutionContext`.  
+For instance:
+
+    ExecutionContext.fromExecutor(new ThreadPoolExecutor( /* your configuration */ ))
+
+
+### Synchronous Execution Context
+
+One might be tempted to have an `ExecutionContext` that runs computations within the current thread:
+
+    val currentThreadExecutionContext = ExecutionContext.fromExecutor(
+      new Executor {
+        def execute(runnable: Runnable) { runnable.run() }
+    })
+
+This should be avoided as it introduce non-determinism in the execution of your future.
+
+    Future { doSomething }(ExecutionContext.global).map { doSomethingElse }(currentThreadExecutionContext)
+
+`doSomethingElse` might either execute in `doSomething`'s thread or in the main thread, and therefore be either asynchronous or synchronous.
+As explained [here](http://blog.ometer.com/2011/07/24/callbacks-synchronous-and-asynchronous/) a callback shouldn't be both.
```

Comment:
Even though the blogger is in this case an active Scala community member, I'd argue against linking information to blogs, however accurate it may be.

Instead, I'd first open a documentation section in ScalaDocs about concurrency and/or programming paradigm terms in general, and then link that (since, we, as a Scala community then get to maintain it and ensure that it is current and accurate).